client/sites/hooks/test/use-transfer-with-software-start-mutation.tsx

```diff
@@ -0,0 +1,96 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
+import { renderHook, waitFor } from '@testing-library/react';
+import nock from 'nock';
+import React from 'react';
+import { useRequestTransferWithSoftware } from '../use-transfer-with-software-start-mutation';
+
+const replyErrorWithEnvelope =
+	( status: number, defaultBody: Record< string, string | number > = {} ) =>
+	( body = {} ) =>
+	() => [ 200, { code: status, body: { ...defaultBody, ...body } } ];
+const errorResponse = replyErrorWithEnvelope( 400, { error: 'any error' } );
+const SITE_ID = 123;
+const PLUGINS = { 'plugin-1': 'install' as const };
+const THEMES = { 'theme-1': 'activate' as const };
+
+const Wrapper =
+	( queryClient: QueryClient ) =>
+	( { children } ) => (
+		<QueryClientProvider client={ queryClient }>{ children }</QueryClientProvider>
+	);
+
+const render = ( options = { retry: 0 } ) => {
+	const queryClient = new QueryClient();
+
+	const renderResult = renderHook(
+		() => useRequestTransferWithSoftware( SITE_ID, PLUGINS, THEMES, options ),
+		{
+			wrapper: Wrapper( queryClient ),
+		}
+	);
+
+	return {
+		...renderResult,
+		queryClient,
+	};
+};
+
+describe( 'useRequestTransferWithSoftware', () => {
+	beforeAll( () => {
+		nock.disableNetConnect();
+	} );
+
+	beforeEach( () => nock.cleanAll() );
+
+	it( 'should successfully request transfer with software and return the transferId', async () => {
+		nock( 'https://public-api.wordpress.com' )
+			.post( '/rest/v1.1/sites/' + SITE_ID + '/atomic/transfer-with-software', {
```

Comment:
This is pointing at the wrong endpoint. Using the correct one does not work because the nock requests don't match; something in the middleware seems to be defaulting to `rest/v1.1`

Reply:
Yes, it is probably caused by our wpcom lib. 
Unfortunately, it works differently on node env.

Maybe for now, we can just leave a comment on the code explaining what the correct endpoint is, and then we can create an issue to investigate the wpcom.

Reply:
Sigh, I figured it out. I had the `apiNamespace` in the wrong place. Fixed!